lib/gelfx/log_entry.ex

```diff
@@ -185,4 +183,31 @@ defmodule Gelfx.LogEntry do
   def short_message(<<>>, original, _length) do
     original
   end
+
+  defp formatted_full_message({module, func}, level, message, timestamp, metadata) do
+    try do
+      with true <- Code.ensure_loaded?(module),
+           true <- function_exported?(module, func, 4) do
+        apply(module, func, [level, message, timestamp, metadata])
+      else
+        _ ->
+          default_formatted_full_message(level, message, timestamp, metadata)
+      end
+    rescue
+      _ ->
+        default_formatted_full_message(level, message, timestamp, metadata)
+    end
+  end
+
+  defp formatted_full_message(format, level, message, timestamp, metadata) do
+    format
+    |> Formatter.compile()
```

Comment:
I would move this somewhere in the backend initialization.
Like this the format is compiled every time a message is send.